fairlearn/metrics/_metrics_engine.py

```diff
@@ -167,3 +261,66 @@ def group_max_from_summary(summary):
 def _check_array_sizes(a, b, a_name, b_name):
     if len(a) != len(b):
         raise ValueError(_MESSAGE_SIZE_MISMATCH.format(b_name, a_name))
+
+
+TRANSFORMATIONS = {
+    "difference": difference_from_summary,
+    "ratio": ratio_from_summary,
+    "group_min": group_min_from_summary,
+    "group_max": group_max_from_summary,
+}
+
+# Base metrics and the variants that are implemented by the metrics engine
+METRICS_SPEC = [
+    # base metrics from _extra_metrics
+    (true_positive_rate, ["difference", "ratio"]),
+    (true_negative_rate, []),
```

Comment:
Why no difference and ratio?

Reply:
This follows the API proposal. The only reason I ended up expanding `true_positive_rate` and `false_positive_rate` is because they are used to define `equalized_odds_{difference,ratio}`. There is also a separate rationale for `true_positive_rate` since it correponds to the "equal opportunity" metric.

Beyond that, I tried to keep the number of automatically expanded metrics to the minimum. It will be easy to add new derived metrics in future, but we won't be able to remove metnics easily. So I'm trying to start with the "minimal sensible set" and expand based on the usage pattern.

Reply:
OK